node/src/reactor/main_reactor/control.rs

```diff
@@ -471,16 +475,17 @@ impl MainReactor {
     }
 
     pub(super) fn should_commit_upgrade(&self) -> bool {
-        let highest_switch_block_header = match &self.switch_block_header {
-            None => {
-                return false;
+        match &self.storage.read_highest_switch_block_headers(1).as_deref() {
```

Comment:
Same comment as the one from `commit_upgrade` applies here since this check guards the entry into that method. We need to have a complete block that is the last block before activation in order to commit the upgrade.